web/ui/src/DeployWiz/DeployWizard.js

```diff
@@ -328,61 +283,46 @@
+                poolID: $scope.install.poolID,
+                TemplateID: $scope.install.templateID,
+                DeploymentID: $scope.install.deploymentID
+            };
 
-                var checkStatus = true;
-                resourcesFactory.deploy_app_template(deploymentDefinition, function() {
-                    servicesFactory.update().then(function(){
-                        checkStatus = false;
-                        closeModal();
-                    });
-                }, function(){
+            var checkStatus = true;
+            resourcesFactory.deploy_app_template(deploymentDefinition, function() {
+                servicesFactory.update().then(function(){
                     checkStatus = false;
                     closeModal();
                 });
+            }, function(){
+                checkStatus = false;
+                closeModal();
+            });
 
-                //now that we have started deploying our app, we poll for status
-                var getStatus = function(){
-                    if(checkStatus){
-                        var $status = $("#deployStatusText");
-                        resourcesFactory.get_deployed_templates(deploymentDefinition, function(data){
-                            if(data.Detail === "timeout"){
-                                $("#deployStatus .dialogIcon").fadeOut(200, function(){$("#deployStatus .dialogIcon").fadeIn(200);});
+            //now that we have started deploying our app, we poll for status
+            var getStatus = function(){
+                if(checkStatus){
+                    var $status = $("#deployStatusText");
+                    resourcesFactory.get_deployed_templates(deploymentDefinition, function(data){
+                        if(data.Detail === "timeout"){
+                            $("#deployStatus .dialogIcon").fadeOut(200, function(){$("#deployStatus .dialogIcon").fadeIn(200);});
+                        }else{
+                            var parts = data.Detail.split("|");
+                            if(parts[1]){
+                                $status.html('<strong>' + $translate.instant(parts[0]) + ":</strong> " + parts[1]);
                             }else{
-                                var parts = data.Detail.split("|");
-                                if(parts[1]){
-                                    $status.html('<strong>' + $translate.instant(parts[0]) + ":</strong> " + parts[1]);
-                                }else{
-                                    $status.html('<strong>' + $translate.instant(parts[0]) + '</strong>');
-                                }
+                                $status.html('<strong>' + $translate.instant(parts[0]) + '</strong>');
                             }
+                        }
 
-                            getStatus();
-                        });
-                    }
-                };
+                        getStatus();
```

Comment:
I don't see any kind of delay or timeout between polls here. Am I missing something?

Also kind of confusing - get_deployed_templates checks on the deployment status, but this callback doesn't clear `checkStatus`. When does the polling stop?